tools/benchmark/run.go

```diff
@@ -59,6 +59,43 @@ func makeHTTPRequest(ctx context.Context, u *url.URL, msg schema.Message) (*http
 	return req, nil
 }
 
+type Limiter struct {
```

Comment:
this looks kinda like https://pkg.go.dev/golang.org/x/time/rate

want to try to use that instead? the previous impl was mainly chosen for simplicity but if we're not getting results from it, using something more substantial may be the best option.

That said, what's your theory as to why 429s are occurring? is it overloading at the start of execution? is it steady state excesses? I suspect it's mostly the latter in which case you might try a token bucket construct. if it's the former, you'd probably want to pair that with a slower initial token bucket fill.